src/ucp/proto/proto_am.inl

```diff
@@ -675,10 +670,7 @@ ucp_am_bcopy_handle_status_from_pending(uct_pending_req_t *self, int multi,
     ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
 
     if (ucs_unlikely(status == UCS_ERR_NO_RESOURCE)) {
-        status = ucp_am_copy_header_to_internal_buf(req);
-        if (ucs_likely(status == UCS_OK)) {
-            status = UCS_ERR_NO_RESOURCE;
-        }
+        status = ucp_am_copy_header_to_internal_buf(req, status);
     }
 
     return ucp_bcopy_handle_status_from_pending(self, multi, tag_sync, status);
```

Comment:
IMO it shold be separate `ucp_am_bcopy_handle_status_from_pending`, and `ucp_am_copy_header_to_internal_buf` should not handle `status` as an input parameter

Reply:
It's can make some confusion, but you can look on ucp_am_copy_header_to_internal_buf as another pending handler for the UCS_ERR_NO_RESOURCE case it's make some sense.
It's also making the code much cleaner.
Because ucp_am_copy_header_to_internal_buf is called only on request with status UCS_ERR_NO_RESOURCE Maybe we can return UCS_ERR_NO_RESOURCE if it's succeed and UCS_ERR_NO_MEMORY when it's failed?

This way it's will not get status as parameter and the code will remain clean?

Reply:
I've decided to leave the current implementation. 
I don't want to check the return value outside of the copy header function, We already doing that in the copy header function itself when calling for mpool_get func, so it's will add a redundant branch.

Maybe we need to come up with a better name but I want to see what other people thinks.

Reply:
I've left the implementation as is.
IMO the alternative is to handle the return value of the copy header outside of the copy function.
it's will add variable for the return value and "ucs_unlikely" branch.

I want to see what you and other people thinks.

*I think returning UCS_ERR_NO_RESOURCE when this function succeed it's odd so I ignore this option.